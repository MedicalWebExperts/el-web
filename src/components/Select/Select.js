import React, { Component } from 'react';
import { arrayOf, shape, func } from 'prop-types';

import theme from '../../styles';

class Select extends Component {
  static propTypes = {
    options: arrayOf(shape({})).isRequired,
    onChange: func.isRequired,
  };

  state = {
    selectedValue: '',
  };

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ selectedValue: value });
    this.props.onChange({ [name]: value });
  };

  render() {
    const { options } = this.props;
    const { selectedValue } = this.state;
    return (
      <div>
        {
          <style jsx>
            {`
              select {
                padding: 10px;
                font-size: 14px;
                border-radius: 4px;
                border-width: 1px;
                border-style: solid;
                background-color: ${theme.colors.white};
                color: ${theme.colors.textTertiary};
                border-color: ${theme.colors.textTertiary};
                outline: none;
                height: 40px;
                min-width: 160px;
              }
              option {
                color: ${theme.colors.textTertiary};
              }
            `}
          </style>
        }
        <select value={selectedValue} onChange={this.handleChange}>
          {options.map(({ id, name, value }) => (
            <option value={value} key={id}>
              {name}
            </option>
          ))}
        </select>
      </div>
    );
  }
}

export default Select;
