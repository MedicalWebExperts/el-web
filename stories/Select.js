import React from 'react';
import { storiesOf } from '@storybook/react';

import { Select } from '../src';

const options = [
  { name: 'Tiger', value: 'tiger' },
  { name: 'Cheetah', value: 'cheetah' },
  { name: 'Giraffe', value: 'giraffe' },
  { name: 'Monkey', value: 'monkey' },
  { name: 'Wolf', value: 'wolf' },
  { name: 'Duck', value: 'duck' },
];

export default () => {
  storiesOf('Select', module).add('Default', () => (
    <Select onChange={() => null} options={options} placeholder="Animals" />
  ));
};
