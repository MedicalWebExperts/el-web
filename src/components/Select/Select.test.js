import React from 'react';
import renderer from 'react-test-renderer';

import Select from './Select';

const options = [
  { name: 'Tiger', value: 'tiger' },
  { name: 'Cheetah', value: 'cheetah' },
  { name: 'Giraffe', value: 'giraffe' },
  { name: 'Monkey', value: 'monkey' },
  { name: 'Wolf', value: 'wolf' },
  { name: 'Duck', value: 'duck' },
];

const props = {
  name: 'select',
  placeholder: 'this is a placeholder',
  onChange: () => null,
  options,
};

const customStyles = {
  marginTop: 30,
  lineHeight: 2,
};

describe('Component Select Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Select {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const tree = renderer.create(<Select {...props} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
