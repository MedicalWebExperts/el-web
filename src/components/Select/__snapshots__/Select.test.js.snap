// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Select Snapshot added optional style inline prop 1`] = `
<div
  className="jsx-2953334337"
>
  <select
    className="jsx-2953334337"
    onChange={[Function]}
    style={
      Object {
        "lineHeight": 2,
        "marginTop": 30,
      }
    }
    value=""
  >
    <option
      className="jsx-2953334337"
      default={true}
      disabled={true}
      value=""
    >
      this is a placeholder
    </option>
    <option
      className="jsx-2953334337"
      value="tiger"
    >
      Tiger
    </option>
    <option
      className="jsx-2953334337"
      value="cheetah"
    >
      Cheetah
    </option>
    <option
      className="jsx-2953334337"
      value="giraffe"
    >
      Giraffe
    </option>
    <option
      className="jsx-2953334337"
      value="monkey"
    >
      Monkey
    </option>
    <option
      className="jsx-2953334337"
      value="wolf"
    >
      Wolf
    </option>
    <option
      className="jsx-2953334337"
      value="duck"
    >
      Duck
    </option>
  </select>
</div>
`;

exports[`Component Select Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-2953334337"
>
  <select
    className="jsx-2953334337"
    onChange={[Function]}
    style={Object {}}
    value=""
  >
    <option
      className="jsx-2953334337"
      default={true}
      disabled={true}
      value=""
    >
      this is a placeholder
    </option>
    <option
      className="jsx-2953334337"
      value="tiger"
    >
      Tiger
    </option>
    <option
      className="jsx-2953334337"
      value="cheetah"
    >
      Cheetah
    </option>
    <option
      className="jsx-2953334337"
      value="giraffe"
    >
      Giraffe
    </option>
    <option
      className="jsx-2953334337"
      value="monkey"
    >
      Monkey
    </option>
    <option
      className="jsx-2953334337"
      value="wolf"
    >
      Wolf
    </option>
    <option
      className="jsx-2953334337"
      value="duck"
    >
      Duck
    </option>
  </select>
</div>
`;
